expo/ticketing/models.py

```diff
@@ -146,7 +157,167 @@ class Transaction(models.Model):
+                    profile,
+                    conference))
+        return no_exclusion
+
+
+class TicketingEligibilityCondition(EligibilityCondition):
+    '''
+    This is the implementation of the condition under which we give a
+    checklist item to a purchaser because they have purchased a ticket
+    Tickets are realized as BPT Events, the various tickets within an
+    event do not qualify a user for anything more.
+
+    Ticket conditions are additive.  X purchases = X items given to the
+    buyer
+    '''
+    tickets = models.ManyToManyField(TicketItem,
+                                     blank=False)
+
+    def __unicode__(self):
+        unicode_string = ""
+        first = True
+        for ticket in self.tickets.all():
+            if first:
+                unicode_string = unicode(ticket)
+                first = False
+            else:
+                unicode_string += ", " + unicode(ticket)
+
+        return unicode_string
+
+
+class RoleEligibilityCondition(EligibilityCondition):
+    '''
+    This is the implementation of the condition under which we give a
+    checklist item to a person because they fulfill an assigned role.
+
+    Roles are given once per person per conference - being a role
+    gets exactly 1 of the item.
+    '''
+    role = models.CharField(max_length=25,
+                            choices=role_options)
+
+    def __unicode__(self):
+        return unicode(self.role)
+
+
+class Exclusion(models.Model):
+    '''
+    This is the abstract class connecting the cases under which a
+    CheckListItem should be assigned, even when it meets the current
+    condition.  Exclusions are combined in many to 1 conditions as
+    logical OR cases - any case being true negates the condition.
+    '''
+    condition = models.ForeignKey(
+        EligibilityCondition,
+        related_name="%(app_label)s_%(class)s")
+
+    class Meta:
+        abstract = True
+
+
+class TicketingExclusion(Exclusion):
+    '''
+    This is the implementation of the case when the presence of a ticket
+    purchase eliminates the eligibility of the individual for getting the
+    checklist item under the given condition.  The usual case is that a
+    person purchased a ticket that includes an equivalent to the current
+    item and maybe more.
+    '''
+    tickets = models.ManyToManyField(TicketItem,
+                                     blank=False)
+
+    def __unicode__(self):
+        unicode_string = ""
+        first = True
+        for ticket in self.tickets.all():
+            if first:
+                unicode_string = unicode(ticket)
+                first = False
+            else:
+                unicode_string += ", " + unicode(ticket)
+
+        return unicode_string
+
+    def is_excluded(self, held_tickets):
+        no_exclusion = True
+        for ticket in held_tickets:
+            if ticket in self.tickets.all():
+                no_exclusion = False
+
+        return no_exclusion
```

Comment:
I'm not sure about the logic here. is_excluded returns True unless some ticket in held_tickets is in this exclusion's ticket set? I think you want to return `not no_exclusion`, unless I'm missing something